### ISSUE TITLE
Integrate OpenSearch Dashboards release functional tests into CI process

### ISSUE THREAD

**Issue by tianleh** [opensearch-project/opensearch-build#604]
### Is your feature request related to a problem? Please describe

We want to integrate OSD release test https://github.com/opensearch-project/opensearch-dashboards-functional-test into the infra CI. We have already verified that it can run in Github Action env.

(See this sample run https://github.com/opensearch-project/opensearch-dashboards-functional-test/runs/3705023275?check_suite_focus=true)

### Describe the solution you'd like

The overall process would be:

For security enabled cluster,

    checkout opensearch-dashboards-functional-test from Github
    run npx cypress run --spec "cypress/integration/security-enabled/*.js"

For security disabled cluster,

    checkout opensearch-dashboards-functional-test from Github
    run npx cypress run --spec "cypress/integration/security-disabled/*.js"

(One call out is that opensearch-dashboards-functional-test imports another library repo https://github.com/opensearch-project/opensearch-dashboards-test-library through NPM registry.)

### Describe alternatives you've considered

We could just use the Github Actions https://github.com/opensearch-project/opensearch-dashboards-functional-test However, it is isolated outside of the org workflow.

### Task breakdown

- [x] https://github.com/opensearch-project/opensearch-build/issues/834
- [x] https://github.com/opensearch-project/opensearch-build/issues/852
- [x] https://github.com/opensearch-project/opensearch-build/issues/837
    - [x] https://github.com/opensearch-project/opensearch-build/issues/1079
    - [x] https://github.com/opensearch-project/opensearch-build/issues/1080
    - [x] https://github.com/opensearch-project/opensearch-build/issues/1134
    - [x] https://github.com/opensearch-project/opensearch-build/issues/1081
- [x] https://github.com/opensearch-project/opensearch-build/issues/838
    - [x] https://github.com/opensearch-project/opensearch-build/issues/1161
    - [x] https://github.com/opensearch-project/opensearch-build/issues/1170
    - [x] https://github.com/opensearch-project/opensearch-build/issues/1171
    - [x] https://github.com/opensearch-project/opensearch-build/issues/1177
    - [x] https://github.com/opensearch-project/opensearch-build/issues/1228
    - [x] https://github.com/opensearch-project/opensearch-build/issues/1211
- [x] Jenkins integration
    - [x] https://github.com/opensearch-project/opensearch-build/issues/1259
    - [x] https://github.com/opensearch-project/opensearch-build/issues/1264
    - [x] https://github.com/opensearch-project/opensearch-build/issues/1269
    - [x] https://github.com/opensearch-project/opensearch-build/issues/1322
    - [x] https://github.com/opensearch-project/opensearch-build/issues/1323
    - [x] https://github.com/opensearch-project/opensearch-build/issues/1371
    - [x] https://github.com/opensearch-project/opensearch-build/issues/1372
    - [x] https://github.com/opensearch-project/opensearch-build/issues/1373
    - [x] https://github.com/opensearch-project/opensearch-build/issues/1381
    - [x] https://github.com/opensearch-project/opensearch-build/issues/1482
    - [x] https://github.com/opensearch-project/opensearch-build/issues/1489
- [x] Update functional test repo
    - [x] https://github.com/opensearch-project/opensearch-dashboards-functional-test/issues/44
    - [x] https://github.com/opensearch-project/opensearch-dashboards-functional-test/issues/39
    - [x] https://github.com/opensearch-project/opensearch-dashboards-functional-test/issues/49

Fast Followups
- [x] https://github.com/opensearch-project/opensearch-build/issues/1190
- [x] https://github.com/opensearch-project/opensearch-build/issues/1191
- [x] https://github.com/opensearch-project/opensearch-build/issues/1707
- [x] https://github.com/opensearch-project/opensearch-build/issues/1708
- [x] https://github.com/opensearch-project/opensearch-build/issues/1709
- [ ] https://github.com/opensearch-project/opensearch-build/issues/1728

**Comment by peterzhuamazon** [comment]
Since test-workflow haven't officials support dashboards/cypress yet, we will try to run this on the old Jenkins Workflow next week with repository integtest.sh.
We will migrate it to test-workflow once it is ready.

Thanks.

**Comment by seraphjiang** [comment]
what's the gap to run cypress? is there an open issue to track, want to know more

**Comment by peterzhuamazon** [comment]
We have successfully test the changes on Jenkins with both Security and No-Security tests on x64 docker container.
Due to arm64 cypress not having available binary, we have not test on arm64 yet.

**Comment by seraphjiang** [comment]
Is there an issue to track we need two parameters need to run test?

**Comment by peternied** [comment]
[Triage] @tianleh Could you add individual tasks that would align to the PRs that you are creating where this task is broken into more digestible pieces?

**Comment by peternied** [comment]
[Triage] @peterzhuamazon is tracking this item, and the costing reflects this.  If it makes sense to break this apart into more than one item please do what works best for you.

**Comment by tianleh** [comment]
Task breakdown

- [x] https://github.com/opensearch-project/opensearch-build/issues/834
- [ ] https://github.com/opensearch-project/opensearch-build/issues/852
- [ ] https://github.com/opensearch-project/opensearch-build/issues/837
- [ ] https://github.com/opensearch-project/opensearch-build/issues/838

**Comment by tianleh** [comment]
My POC work https://github.com/tianleh/opensearch-build/tree/add-test-support is working. Focusing on moving the POC code to official repo

**Comment by tianleh** [comment]
Added more subtasks under https://github.com/opensearch-project/opensearch-build/issues/837 to make incremental progress. The update is in the top section of this issue. ("Is your feature request related to a problem? Please describe")

**Comment by tianleh** [comment]
To be able to create work parallelization among CI/CD engineers, I am thinking about creating new Github issues for some code optimization/enhancement feedback received from PR review (instead of directly addressing them as part of its originating PR).

Then my engineering capacity won't be the bottleneck of the launch of the whole task #604 . To ensure ownership of those new issues, I will be the first owner and then coordinate among the team for the actual implementation work.

The leadership team is expecting the Dashboards CI (plugin as well) to be complete (GA) by 12/15/2021 Thus for those new Github issues, we will also define their priority as pre GA or post GA and focus on pre GA ones.

@seraphjiang @dblock @peternied @kavilla

**Comment by dblock** [comment]
> To be able to create work parallelization among CI/CD engineers, I am thinking about creating new Github issues for some code optimization/enhancement feedback received from PR review (instead of directly addressing them as part of its originating PR).
> 
> Then my engineering capacity won't be the bottleneck of the launch of the whole task #604 . To ensure ownership of those new issues, I will be the first owner and then coordinate among the team for the actual implementation work.
> 
> The leadership team is expecting the Dashboards CI (plugin as well) to be complete (GA) by 12/15/2021 Thus for those new Github issues, we will also define their priority as pre GA or post GA and focus on pre GA ones.
> 
> @seraphjiang @dblock @peternied @kavilla

If a piece of feedback on a PR is not a must have and is being deferred because it's too much work for the PR, opening an issue for that is the right thing to do so we don't forget, anyway.

**Comment by tianleh** [comment]
To track PR/Issue related to release integ test, I created a new label `ci-test-automation`

To track those PRs which have been approved/merged with comments, I created a new label `comment-to-followup`
(Once remaining comments are resolved in follow up PRs, I will remove the label)

**Comment by tianleh** [comment]
Added a few items to track our work with infra team about upgrading to Chromium to 
1) stabalize the test
2) support Reporting plugin

**Comment by bbarani** [comment]
@tianleh Whats the status of this issue?

**Comment by bbarani** [comment]
@tianleh @peterzhuamazon @kavilla Can you please update the current status of this issue?

**Comment by gaiksaya** [comment]
Closing this issue as we are now using opensearch-functional-test repo to run tests for all dashboards plugin. No action items pending from infra side.

Please re-open if you think any items are pending. 

Thanks!
